### PR TITLE
docs(cicd): update docs

### PR DESCRIPTION
**Pull request by warren830**
# Summary

1. Clarify what GitHub/Jenkins entities are filtered by the deployment regex #https://github.com/apache/incubator-devlake/issues/4313

2. Add cicd_scopes in domain layer schema

### Does this close any open issues?
Closes issue https://github.com/apache/incubator-devlake/issues/4313

### CODE REVIEW

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 1]
```diff
@@ -435,6 +435,20 @@ Events of pull requests.
 
 ### Domain 4 - CI/CD(WIP)
 
+#### cicd_scopes
+
+Information about Jenkins Job, GitHub Action or Gitlab CI. 
+
+| **field**      | **type** | **length** | **description**                                                                                                                                                                                                | **key**        |
+| :------------- | :------- | :--------- |:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| :------------- |
+| `id`           | varchar  | 255        | A cicd_scope's `id` is composed of "< plugin >:< Entity >:< PK0 >[:PK1]..."<br/>For example, a Github cicd_scope's id is like "< github >:< GithubRepos >< GithubRepoId >". Eg. 'github:GithubRepos:384111310' | PK             |
```
Comment (hezyin) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: I thought the connection id is also concatenated into the id?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 2]
```diff
@@ -435,6 +435,20 @@ Events of pull requests.
 
 ### Domain 4 - CI/CD(WIP)
 
+#### cicd_scopes
+
+Information about Jenkins Job, GitHub Action or Gitlab CI.
```
Comment (hezyin) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: This one-line description is a bit too abstract. Maybe it's easier for users to understand if we explain cicd_scopes as a collection of cicd_pipelines and give some examples.

For GitHub: a GitHub repo is converted to a cicd_scope
For Jenkins: <I'm actually not sure the Jenkins entity that corresponds to a cicd_scope>
For GitLab: I assume it's also a GitLab repo?
Any other data source that may produce cicd_scope?

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 3]
```diff
@@ -435,6 +435,20 @@ Events of pull requests.
...
+| `description`  | varchar  | 255        | The description of cicd_scope.                                                                                                                                                                                 |                |
+| `url`          | varchar  | 255        | The url of cicd_scope. Eg. https://github.com/apache/incubator-devlake or https://jenkins.xxx.cn/view/PROD/job/OPS_releasev2/                                                                                  |                |
+| `created_date` | datetime | 3          | cicd_scope creation date                                                                                                                                                                                       |                |
+| `updated_date` | datetime | 3          | Last full update was done for this cicd_scope                                                                                                                                                                  |                |
```
Comment (hezyin) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: "Last full update was done for this cicd_scope" -> "Date of the last data collection for this cicd_scope"

Is emphasizing "full update" important here?

File: docs/UserManuals/ConfigUI/GitHub.md [hunk 4]
```diff
@@ -120,6 +120,13 @@ If you're using GitHub Action to conduct `deployments`, please select "Detect De
 - Deployment: A GitHub Action job with a name that matches the given regEx will be considered as a deployment.
 - Production: A GitHub Action job with a name that matches the given regEx will be considered a job in the production environment.
 
+A GitHub workflow run has many jobs. Each GitHub workflow run is converted to a
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/GitHub.md:123]: "A GitHub workflow run has many jobs" -> "A GitHub workflow has many jobs" (remove the "run")

File: docs/UserManuals/ConfigUI/GitHub.md [hunk 5]
```diff
@@ -120,6 +120,13 @@ If you're using GitHub Action to conduct `deployments`, please select "Detect De
 - Deployment: A GitHub Action job with a name that matches the given regEx will be considered as a deployment.
 - Production: A GitHub Action job with a name that matches the given regEx will be considered a job in the production environment.
 
+A GitHub workflow run has many jobs. Each GitHub workflow run is converted to a 
+cicd_pipeline in the domain layer and each GitHub Action job is converted to a cicd_task in the domain layer.
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/GitHub.md:124]: "each GitHub Action job" -> "each GitHub Actions job run"

File: docs/UserManuals/ConfigUI/GitHub.md [hunk 6]
```diff
@@ -120,6 +120,13 @@ If you're using GitHub Action to conduct `deployments`, please select "Detect De
...
+![github-action-run](/img/ConfigUI/github-action-run.png)
+![github-action-job](/img/ConfigUI/github-action-job.png)
+
+The deployment and production regex is always applied to the records in the cicd_tasks table.
+
 By the above two fields, DevLake can identify a production deployment among massive CI jobs.
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/GitHub.md]: "The regexes allow users to filter their CI/CD pipelines for deployments (to production)."

File: versioned_docs/version-v0.15/DataModels/DevLakeDomainLayerSchema.md [hunk 7]
```diff
@@ -435,6 +435,20 @@ Events of pull requests.
 
 ### Domain 4 - CI/CD(WIP)
 
+#### cicd_scopes
```
Comment (hezyin) [inline versioned_docs/version-v0.15/DataModels/DevLakeDomainLayerSchema.md:438]: Please replicate the changs to the v0.15 doc as well.

File: versioned_docs/version-v0.15/UserManuals/ConfigUI/GitHub.md [hunk 8]
```diff
@@ -120,6 +120,12 @@ If you're using GitHub Action to conduct `deployments`, please select "Detect De
 - Deployment: A GitHub Action job with a name that matches the given regEx will be considered as a deployment.
 - Production: A GitHub Action job with a name that matches the given regEx will be considered a job in the production environment.
 
+A GitHub workflow run has many jobs. Each GitHub workflow run is converted to a cicd_pipeline in the domain layer and each GitHub Action job is converted to a cicd_task in the domain layer.
```
Comment (hezyin) [inline versioned_docs/version-v0.15/UserManuals/ConfigUI/GitHub.md:123]: Please replicate the changs to the v0.15 doc as well.

File: versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md [hunk 9]
```diff
@@ -51,14 +51,22 @@ Jenkins only supports `CI/CD` domain entities, transformed from Jenkins builds a
 - CI/CD: Jenkins builds, stages, etc.
 
 ### Step 3 - Adding Transformation Rules (Optional)
-
+Before you start, you should understand that
```
Comment (hezyin) [inline versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md]: Why is this sentence here?

File: versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md [hunk 10]
```diff
@@ -51,14 +51,22 @@ Jenkins only supports `CI/CD` domain entities, transformed from Jenkins builds a
...
 
 ### Step 3 - Adding Transformation Rules (Optional)
-
+Before you start, you should understand that 
 This set of configurations is used for calculating [DORA metrics](../DORA.md).
 
 If you're using Jenkins builds to conduct `deployments`, please select "Detect Deployment from Jenkins Builds", and input the RegEx in the following fields:
 
-- Deployment: A Jenkins build with a name that matches the given regEx will be considered as a deployment.
+- Deployment: A Jenkins build with its job name or a Jenkins build stage with its name that matches the given regEx will be considered as a deployment.
```
Comment (hezyin) [inline versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md]: Please replicate the changs to the v0.15 doc as well.

File: docs/UserManuals/ConfigUI/Jenkins.md [hunk 11]
```diff
@@ -59,6 +59,14 @@ If you're using Jenkins builds to conduct `deployments`, please select "Detect D
 - Deployment: A Jenkins build with a name that matches the given regEx will be considered as a deployment.
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/Jenkins.md]: I recommend revising to below:

If you'd like to define `deployments` with Jenkins, please select "Detect Deployment from Jenkins Builds", and provide the following regexes

- Deployment: Jenkins stages whose names match the regex will be registered as deployments (if a Jenkins job has no stage, its job name will be used to match the regex)
- Production: Jenkins stages whose names match the regex will be assigned environment 'PRODUCTION' (if a Jenkins job has no stage, its job name will be used to match the regex)

File: docs/UserManuals/ConfigUI/Jenkins.md [hunk 12]
```diff
@@ -59,6 +59,14 @@ If you're using Jenkins builds to conduct `deployments`, please select "Detect D
...
 - Production: A Jenkins build with a name that matches the given regEx will be considered a build in the production environment.
 
+There're two cases. If a Jenkins build has no stage, it's converted to both a cicd_task and a cicd_pipeline,
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/Jenkins.md]: This is how it works behind the scene:

- If a Jenkins job has stages, it's converted to a cicd_pipeline and its stages are converted to cicd_tasks in DevLake's domain layer schema.
- If a Jenkins job has no stage, it's converted to both a cicd_pipeline and a cicd_task whose names are the job's name.

After the conversion, the two regexes are applied to the records in the cicd_tasks table.

File: docs/UserManuals/ConfigUI/Jenkins.md [hunk 13]
```diff
@@ -59,6 +59,14 @@ If you're using Jenkins builds to conduct `deployments`, please select "Detect D
...
+both have the same name as the Jenkins job's name.
+If a Jenkins build has stages, it's converted to a cicd_pipeline and its
+stages are converted to cicd_tasks in the domain layer.
+![jenkins-job-build-stage](/img/ConfigUI/jenkins-job-build-stage.png)
+
+The deployment and production regex is always applied to the records in the cicd_tasks table.
+
 By the above two fields, DevLake can identify a production deployment among massive CI jobs.
```
Comment (hezyin) [inline docs/UserManuals/ConfigUI/Jenkins.md]: We can remove this line.

**Review by hezyin — CHANGES_REQUESTED** [review]
Good work, please see my comments.

File: docs/UserManuals/ConfigUI/GitHub.md (hunk 4 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/GitHub.md:123]: But run is an execution, workflow is just a definition? So workflow run is converted to cicd_pipeline. Will it be ambiguous if we remove run?

File: docs/DataModels/DevLakeDomainLayerSchema.md (hunk 2 above)
Comment (warren830) [reply docs/DataModels/DevLakeDomainLayerSchema.md]: fixed, we also have bitbutcket, but I think we don't need to enumerate every plugin?

File: docs/DataModels/DevLakeDomainLayerSchema.md (hunk 3 above)
Comment (warren830) [reply docs/DataModels/DevLakeDomainLayerSchema.md]: `Date of the last data collection for this cicd_scope` is better, hh

File: docs/UserManuals/ConfigUI/GitHub.md (hunk 5 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/GitHub.md:124]: Hmm, but according to github definition, it's just a job.. not a job run

File: docs/UserManuals/ConfigUI/Jenkins.md (hunk 13 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/Jenkins.md]: fixed

File: docs/UserManuals/ConfigUI/Jenkins.md (hunk 12 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/Jenkins.md]: fixed

File: docs/UserManuals/ConfigUI/Jenkins.md (hunk 11 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/Jenkins.md]: fixed

Comment (warren830) [reply docs/UserManuals/ConfigUI/Jenkins.md]: Also, we need to say` if a Jenkins build has no stage`

File: docs/UserManuals/ConfigUI/Jenkins.md (hunk 12 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/Jenkins.md]: `If a Jenkins build has no stage, it's converted to both a cicd_pipeline and a cicd_task whose names are the job's name.

Cause jenkins job is converted to cicd_scope

File: versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md (hunk 10 above)
Comment (warren830) [reply versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md]: fixed

File: versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md (hunk 9 above)
Comment (warren830) [reply versioned_docs/version-v0.15/UserManuals/ConfigUI/Jenkins.md]: fixed

File: docs/DataModels/DevLakeDomainLayerSchema.md (hunk 1 above)
Comment (warren830) [reply docs/DataModels/DevLakeDomainLayerSchema.md]: fixed

File: docs/UserManuals/ConfigUI/GitHub.md (hunk 6 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/GitHub.md]: fixed

File: versioned_docs/version-v0.15/DataModels/DevLakeDomainLayerSchema.md (hunk 7 above)
Comment (warren830) [reply versioned_docs/version-v0.15/DataModels/DevLakeDomainLayerSchema.md:438]: fixed

File: docs/UserManuals/ConfigUI/GitHub.md (hunk 6 above)
Comment (warren830) [reply docs/UserManuals/ConfigUI/GitHub.md]: delete "By the above two fields, DevLake can identify a production deployment among massive CI jobs."

**Review by yumengwang03 — APPROVED** [review]
LGTM

**Review by hezyin — APPROVED** [review]
LGTM

**Pull request merged**